edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.E2E.Test/AgentTests.cs

```diff
@@ -145,7 +143,12 @@ public async Task AgentStartsUpModules(TestConfig testConfig)
                 var dockerCommandFactory = new DockerCommandFactory(client, loggingConfig, configSource.Object, new CombinedDockerConfigProvider(Enumerable.Empty<AuthConfig>()));
                 IRuntimeInfoProvider runtimeInfoProvider = await RuntimeInfoProvider.CreateAsync(client);
                 IEnvironmentProvider environmentProvider = await DockerEnvironmentProvider.CreateAsync(runtimeInfoProvider, restartStateStore, restartManager);
-                var commandFactory = new LoggingCommandFactory(dockerCommandFactory, loggerFactory);
+
+                var logFactoryMock = new Mock<ILoggerFactory>();
```

Comment:
Can you add some details on why this changes is needed?

Reply:
We did the same implementation in LoggingCommandFactoryTest.  Since tests will call ILoggerFactory.CreateLogger, we need to setup to return logger; otherwise the logger will throw NRE when calling Log method.

I don't know why it is working previously, but I guess updating logging NuGet package changed the behavior.

Do you want me to explicitly add comment in the code? it looks weird if only this log mocking code will have comments but not for other usages.

Reply:
No this is good.